saged/tissue_split.py

```diff
@@ -0,0 +1,230 @@
+            with open(args.neptune_config) as neptune_file:
+                neptune_config = yaml.safe_load(neptune_file)
+                neptune_run = utils.initialize_neptune(neptune_config)
+
+        model_type = model_config['name']
+        SupervisedClass = getattr(models, model_type)
+        root_model = SupervisedClass(**model_config)
+
+        pretrain_base = copy.deepcopy(root_model)
+
+        # Get the pretraining fraction of the data
+        pretrain_data.set_label_encoder(label_encoder)
+
+        # Ensure the model is training on GPU if possible
+        if torch.cuda.is_available() and type(pretrain_base) != LogisticRegression:
+            pretrain_base.model = pretrain_base.model.to('cuda')
+        pretrain_base.fit(pretrain_data, neptune_run)
+
+        if neptune_run is not None:
+            neptune_run.stop()
+
+        if issubclass(type(pretrain_base), PytorchSupervised):
+            final_layer_name = pretrain_base.final_layer_name
+            print(final_layer_name)
+            final_layer = getattr(pretrain_base.model, final_layer_name)
+
+            # Reinitialize final layer
+            final_layer.reset_parameters()
+
+        # Move model back to CPU to allow easy copying
+        if type(pretrain_base) != LogisticRegression:
+            pretrain_base.model = pretrain_base.model.to('cpu')
+
+        for i in range(len(labeled_splits)):
+            for subset_number in range(2, 11, 2):
+
+                # Grab a pretrained model and a copy of the original initialization
+                pretrained_model = copy.deepcopy(pretrain_base)
+                no_pretraining_model = copy.deepcopy(root_model)
+
+                # https://github.com/facebookresearch/higher/pull/15
+                gc.collect()
+
+                subset_percent = subset_number * .1
+
+                # Split = 50% pretrain, 40% train, 10% val
+                train_list = labeled_splits[:i] + labeled_splits[i+1:]
+                val_data = labeled_splits[i]
+
+                DatasetClass = type(labeled_data)
+                train_data = DatasetClass.from_list(train_list)
+
+                # Ensure the labels are encoded the same way in all three datasets
+                train_data.set_label_encoder(label_encoder)
+                val_data.set_label_encoder(label_encoder)
+
+                # Now that the ratio is correct, actually subset the samples
+                train_data = train_data.subset_samples(subset_percent,
+                                                       args.seed)
+
+                print('Samples: {}'.format(len(train_data.get_samples())))
+                print('Studies: {}'.format(len(train_data.get_studies())))
+
+                output_size = len(train_data.get_classes())
+
+                print('Val data: {}'.format(len(val_data)))
+                print('output size: {}'.format(output_size))
+
+                # TODO modify save path for no_pretraining_model
```

Comment:
Do you want to keep this TODO here?

Reply:
Yeah, it isn't critical currently, but I'll want to have example models available if/when this goes into a paper